Source/MQTTnet/Client/MqttClient.cs

```diff
@@ -522,7 +530,18 @@ private async Task TryProcessReceivedPacketAsync(MqttBasePacket packet, Cancella
 
                 if (packet is MqttPublishPacket publishPacket)
                 {
-                    await TryProcessReceivedPublishPacketAsync(publishPacket, cancellationToken).ConfigureAwait(false);
+                    if (Options.ParallelReceiving)
+                    {
+                        var task = new Task(async () => await TryProcessReceivedPublishPacketAsync(publishPacket, cancellationToken), cancellationToken);
```

Comment:
Please consider using the ThreadPool.Enqueue.. method because we might get a lot of messages.

Reply:
Do you mean `ThreadPool.QueueUserWorkItem(WaitCallback callBack)`?

I don't know what's the difference between them. But I think the default implementation of TPL may be sufficient.

Another way, `ThreadPool` is not supported in `netstandard1.3`. Shall we consider to upgrade netstandard version?
According to https://docs.microsoft.com/en-us/dotnet/standard/net-standard , I'm not sure we should support the outdated .NET Standard version.

Reply:
Mhh we still have to support netstandard1.3. But I have another question. Why do you need that feature? You can start a new task also in your message handler? What is the benefit in moving that into this library directly?

Reply:
Because of the QOS Ack/Rec

Reply:
@chkr1011 Do you have any idea?

Reply:
I am thinking about a different solution. My idea is to add a new method/property to the event args of the handler. Then you can "disable" the internal auto ACK and do the ACK at a different time. Thats the way how other libs like in Azure do this.  So a new property would indicate the AcknowledgeMode = Manual|Auto and a new "Acknowledge" method will allow the ACK at a different time. If will do nothing for QoS 0 but the proper stuff for QoS 1 and 2.

Reply:
Awesome.
I will also use the parallel receiving in my self project that is a RPC library of MQTT protocol based on MQTTnet.
And in my project, I'm planning 3 options of AcknowledgeMode: OnCall / OnResult / Manual.
It could based on manual acknowledge mode in MQTTnet.

Reply:
It's different with MQTTnet.Extensions.Rpc.
It's based on mqttv5 and broker-side ACL ready, support broadcast and cluster.
It's open-source but still developing.
I'm glad to contribute it into MQTTnet.Extensions.Rpc if you approve it.